central/alert/datastore/internal/store/postgres/bench_test.go

```diff
@@ -39,12 +41,25 @@ func BenchmarkMany(b *testing.B) {
 
 	for n := 1; n < alertsNum; n = n * 2 {
 		b.Run(fmt.Sprintf("upsert %d alerts", n), func(b *testing.B) {
+			startHeap, startHeapObj := getHeapAllocAndObjects(0, 0)
+			var maxHeap uint64
+			var maxHeapObj uint64
+			ticker := time.NewTicker(10 * time.Millisecond)
+			go func() {
+				for range ticker.C {
+					maxHeap, maxHeapObj = getHeapAllocAndObjects(maxHeap, maxHeapObj)
+				}
+			}()
+
 			for b.Loop() {
 				err := store.UpsertMany(ctx, alerts[:n])
 				if err != nil {
 					b.Fatal(err)
 				}
 			}
+			ticker.Stop()
+			b.ReportMetric(float64(maxHeap-startHeap), "max_heap_bytes")
+			b.ReportMetric(float64(maxHeapObj-startHeapObj), "max_heap_objects")
```

Comment:
Looks good, but maybe consider also printing the `startHeap` and `startHeapObj` separately for reference?

Reply:
I'll check that as I remember logging in benchmark breaks benchstat